### ISSUE TITLE
Using set w/ .push on array with nested objects updates existing objects in array AND adds

### ISSUE THREAD

**Issue by RealCK** [mesqueeb/vuex-easy-access#41]
Attempted to use $store.set(Eventsmod/events.push, this.event) from a form and the following:
```
data() {
    return {
      event: {
        date: '',
        time: '',
        title: '',
        description: ''
      },
```
to this module state setup:
```
const state = {
    events:[{
      title:[],
      date:[],
      time:[],
      description:[]
    }]
  };
```
The Payload in vuex dev tool shows that the payload is correct and contains the data, but the store module array does not update correctly. See the state from dev tools here:

Eventsmod:Object
delete:Object (empty)
events:Array[2]
0:Object
date:"2019-07-31"
description:"Test 2"
time:"09:45"
title:"Test 2"
1:Object
date:"2019-07-31"
description:"Test 2"
time:"09:45"
title:"Test 2"
set:Object (empty)

As you can see from the state after 2 pushes, the data is overwritten and added both, I used different data ie. test 1 then test 2, but the last pushed overwrites the previous indexes.  The 2 indexes shown contain the same data. If I push a third with different data, it will show as all 3 having the data contained in the last pushed payload.

I've tried various schema on the module, all arrays(above), events array only with nested objects, just an events array nothing nested.  All have the same outcome.

Additionally, a computed function returns the events using store.get(Eventsmod/events), but it is returning blank objects.  I use that returned data in a vfor loop, which displays a blank tile in the list since it returns blank objects. If you click on the arrow to commit mutation in time travel for the push, the data shows up on the tile. 

Everything works correctly if you pass only a single object to the array.

Any advise is appreciated.

**Comment by mesqueeb** [comment]
Why did you close this?

**Comment by RealCK** [comment]
It was from July of last year and never received a reply, I didn't know what I was doing and this may not even be an actual issue, i think I was doing it wrong.  I still don't know what I'm doing so I closed it assuming a no reply meant I didn't know what I was doing anyway.  I'll buy you a coffee for my annoyance.

**Comment by mesqueeb** [comment]
@RealCK oh i see!! I’m sorry i must have overlooked it last year! I thought you just posted it now. 

Yeah, the question does seem a little hard to follow what exactly the problem is. Do you still need help with this ?
If you want I can have a look at your code.

**Comment by RealCK** [comment]
@mesqueeb I believe I was able to get it working as expected.  If I run into this issue again, I will reopen.  I was not using arrays properly I think.  The only issue I currently face is over on the Vuex Easy Firestore package, which I opened an issue last night regarding.  I'll look for your response over there. Thank You for your help and awesome software.
